juju/juju
# migrations: implement log transfer phase

Streams model logs from the source controller using the debug-log endpoint, and forwards them on to the /migrate/logtransfer endpoint in the target controller.

At the moment this isn't restartable - if the migration worker is killed and restarted while transferring logs the target model will end up with duplicates of all of the records that had already been transferred. This will be added in a follow-up PR.

QA steps: 
* bootstrap a source controller
* add a model to be migrated
* deploy something to the model
* take note of the number of log messages, and the first and last messages: `juju debug-log --replay`
* bootstrap a destination controller
* migrate the model: `juju migrate the-model dest-controller`
* verify that the log messages for the migrated model include all of the expected messages
